### ISSUE TITLE
Code Coverage in SonarQube is much much lower than the coverage in TFS

### ISSUE THREAD

**Issue by wkit23** [SonarSource/sonar-dotnet#1511]
### Description

Code Coverage reported in SonarQube doesn't tally with the Code Coverage reported in TFS.

### Repro steps

1. Setup the build pipeline in TFS with SonarQube.
2. Run the build in TFS.
3. Checks the code coverage in TFS. In our case it is around 62%.
4. Checks the code coverage in SonarQube. It reports 48%.

### Expected behavior

1. SonarQube should report the same Coverage % (or similar) with TFS

### Actual behavior

1. SonarQube reported the code coverage % much lower than TFS.

### Known workarounds

1. None

### Related information

* SonarC# Version 7.2
* TFS 2018 update 2
* Scanner for MS Build 4.3.0
* SonarQube version 6.7.4.38452

I also notice that there're some errors reported in TFS
WARN: Property missing: 'sonar.cs.analyzer.projectOutPaths'. No protobuf files will be loaded for this project.
WARN: No roslyn issues report not found for this project.

And the errors happen for all Unit Tests projects. I have a feeling SonarQube includes the Unit Tests in the coverage calculation as well, and thus explains the difference in the % reported. I also added exclusion in SonarQube to exclude the Unit Tests, but seems it doesn't work.

**Comment by duncanp-sonar** [comment]
@wkit23 [this StackOverflow post](https://stackoverflow.com/questions/49448018/difference-in-codecoverage-report-of-vsts-build-and-sonarqube/49455995#49455995) explains the main reasons why the coverage numbers reported in SonarQube differ from those reported by TFS/VSTS using the default Visual Studio code coverage tooling.

SonarQube doesn't include the unit tests in the coverage calculation but by default the Visual Studio tooling does, which tends to makes the number reported by VS artificially high.
Download the detailed coverage report from TFS, open it in Visual Studio, and drill down into the report to see what the coverage of the product code (rather than the test code) is. You should find that the coverage numbers are much closer to those reported by SonarQube.

The warnings are saying that those projects have not been analysed, so no issues or metrics for those files will be reported (which is what is expected for test projects currently).

**Comment by wkit23** [comment]
Thanks for the link. But however, I think it is the other way around for my case:

- I downloaded the coverage result from the TFS build, and it is clearly DO NOT have any coverage for Unit Tests projects.
- I took a look into the SonarQube project, and I have a feeling that the Coverage % includes the Unit Tests projects. Since the TFS does not have the coverage %, the SonarQube reports 0% and thus brings down the overall coverage result.
- In the previous version (I think it is 5.x) the coverage is correct (at least very similar). This seems to happen after we upgraded to the latest version, and thus is why I think is a bug.
